src/wdio/WdioFileReporter.js

```diff
@@ -0,0 +1,92 @@
+const WDIOSpecReporter = require('wdio-spec-reporter/build/reporter');
+const stripAnsi = require('strip-ansi');
+const fs = require('fs');
+const endOfLine = require('os').EOL;
+const path = require('path');
+const Logger = require('../../scripts/utils/logger');
+
+const LOG_CONTEXT = '[Terra-Toolkit:theme-aggregator]';
+
+class WdioCustomeReporter extends WDIOSpecReporter {
+  constructor(globalConfig, options) {
+    super(globalConfig);
+    this.runners = [];
+    this.resultJsonObject = {
+      startDate: '',
+      type: 'wdio',
+      locale: '',
+      formFactor: '',
+      theme: '',
+      output: [],
+      endDate: '',
+    };
+    if (!options.reporterDir) {
+      this.filePath = path.resolve(__dirname, '..', '..', 'tests/wdio/reports/results');
+    } else {
+      this.filePath = options.reporterDir;
+    }
+    this.fileName = '';
+    this.checkResultDirExist = this.checkResultDirExist.bind(this);
+    this.checkResultDirExist();
+    this.on('runner:end', (runner) => {
+      this.runners.push(runner);
+    });
+  }
+
+  hasReportDir() {
+    const reportDir = path.resolve(this.filePath, '..');
+    if (!fs.existsSync(reportDir)) {
+      fs.mkdirSync(reportDir);
+    }
+  }
+
+  checkResultDirExist() {
+    if (this.filePath && !fs.existsSync(this.filePath)) {
+      this.hasReportDir();
+      fs.mkdirSync(this.filePath);
+    }
+  }
+
+  fileNameCheck() {
+    const { LOCALE = '', THEME = '', FORM_FACTOR = '' } = process.env;
```

Comment:
is it necessary to default these to empty strings if we do an existence check later? if any of these values are undefined, we get the same end result
```suggestion
    const { LOCALE, THEME, FORM_FACTOR } = process.env;
```

Reply:
https://github.com/cerner/terra-toolkit-boneyard/blob/afcb5d15bd7adb1a16a54bdba9797f38b6798188/reporters/wdio/TerraWDIOSpecReporter.js#L51